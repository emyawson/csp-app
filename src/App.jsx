// Modules
import React from 'react';
import { withRouter, BrowserRouter } from 'react-router-dom';

import TagManager from 'react-gtm-module';
import ScrollToTop from './components/scrollToTop/ScrollToTop.jsx';
import Routes from './routers/MainRouter.jsx';
// import MetaTags from './utils/MetaTags';

const tagManagerArgs = {
  gtmId: 'GTM-KQG8LR8',
};
TagManager.initialize(tagManagerArgs);

const App = () => (
    <div className="app">
      <BrowserRouter>
         <ScrollToTop />
         <Routes />
      </BrowserRouter>
    </div>
);

export default withRouter(App);
