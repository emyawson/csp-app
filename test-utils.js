import React from 'react';
import { Provider } from 'react-redux';
import { createStore, combineReducers } from 'redux';
import { Router } from 'react-router-dom';
import { connectRouter } from 'connected-react-router';
import { render } from '@testing-library/react';
import { persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';
import { history } from './src/store';
import reducers from './src/store/reducer/index';

const createRootReducer = (reducerHistory) => combineReducers({
  router: connectRouter(reducerHistory),
  ...reducers,
});

// Setup redux persist
const persistConfig = {
  key: 'root',
  storage,
  stateReconciler: autoMergeLevel2,
};
const persistedReducer = persistReducer(
  persistConfig,
  createRootReducer(history),
);

const renderWithRouter = (ui, options = {}, initialState) => ({
  ...render(
      <Provider store={createStore(persistedReducer, initialState)}>
        <Router history={history}>{ui}</Router>
      </Provider>,
  ),
});

const AllTheProviders = () => {};
const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render, renderWithRouter };
