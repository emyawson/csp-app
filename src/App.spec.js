import React from 'react';
import App from './App.jsx';
import { renderWithRouter } from '../test-utils';
import pagesMock from './services/mock/pagesMock';

const initialState = {
  router: {},
  loading: {},
  pages: pagesMock,
  breadcrumbPathname: [],
  errors: {},
};

const renderComponent = () => renderWithRouter(<App />, {}, initialState);

test('renders <App/>', () => {
  const { container } = renderComponent();
  expect(container).toMatchSnapshot();
});
