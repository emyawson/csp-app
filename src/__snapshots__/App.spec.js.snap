// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <App/> 1`] = `
<div>
  <div
    class="app"
  />
</div>
`;
